test/decoders/test_video_decoder_ops.py

```diff
@@ -322,6 +323,46 @@ def test_frame_pts_equality(self):
             )
             assert pts_is_equal
 
+    @pytest.mark.parametrize("color_conversion_library", ("filtergraph", "swscale"))
+    def test_color_conversion_library(self, color_conversion_library):
+        decoder = create_from_file(str(NASA_VIDEO.path))
+        _add_video_stream(decoder, color_conversion_library=color_conversion_library)
+        frame0, _, _ = get_next_frame(decoder)
+        reference_frame0 = NASA_VIDEO.get_frame_data_by_index(0)
+        assert_tensor_equal(frame0, reference_frame0)
+        reference_frame1 = NASA_VIDEO.get_frame_data_by_index(1)
+        frame1, _, _ = get_next_frame(decoder)
+        assert_tensor_equal(frame1, reference_frame1)
+        seek_to_pts(decoder, 6.0)
+        frame_time6, _, _ = get_next_frame(decoder)
+        reference_frame_time6 = NASA_VIDEO.get_frame_by_name("time6.000000")
+        assert_tensor_equal(frame_time6, reference_frame_time6)
+
+    def test_color_conversion_library_with_scaling(self):
```

Comment:
Just a note: it's worth testing both upscaling and downscaling. Some scaling options like antialising are only relevant when down-scaling.

Also, I guess this isn't the point of this test at this time, but before we enable the `AUTO` strategy to automatically switch between swscale and filtergraph, we should get more coverage of the different resizing options. We need to avoid discrepancies between the resizing strategies of these 2 libraries. I think it's worth mentioning that as a comment in this test.

Reply:
Good callout. I parameterized this test to test upscaling, downscaling and stretching in one dimension while compressing in another.